cmd/autoprowjobdispatcher/main.go

```diff
@@ -0,0 +1,179 @@
+	logrus.Infof("Running: %s", fullCommand)
+	if err := bumper.Call(stdout, stderr, cmd, args...); err != nil {
+		return false, fmt.Errorf("failed to run %s: %w", fullCommand, err)
+	}
+
+	changed, err := bumper.HasChanges()
+	if err != nil {
+		return false, fmt.Errorf("error occurred when checking changes: %w", err)
+	}
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
+	}
+
+	gitCmd := "git"
+	if err := bumper.Call(stdout, stderr, gitCmd, []string{"add", "."}...); err != nil {
+		return false, fmt.Errorf("failed to 'git add .': %w", err)
+	}
+
+	commitArgs := []string{"commit", "-m", fullCommand, "--author", author}
+	if err := bumper.Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
+		return false, fmt.Errorf("fail to %s %s: %w", gitCmd, strings.Join(commitArgs, " "), err)
+	}
+
+	return true, nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.GitHubOptions.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Failed to start secrets agent")
+	}
+
+	gc, err := o.GitHubOptions.GitHubClient(sa, false)
+	if err != nil {
+		logrus.WithError(err).Fatal("error getting GitHub client")
+	}
+
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	steps := []struct {
+		command   string
+		arguments []string
+	}{
+		{
+			command:   "/usr/bin/prow-job-dispatcher",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs", "--prometheus-password-path", o.prometheusPasswordPath, "--prometheus-username", "ci", "--config-path", "core-services/sanitize-prow-jobs/_config.yaml"},
+		},
+		{
+			command:   "/usr/bin/sanitize-prow-jobs",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs", "--config-path", "./core-services/sanitize-prow-jobs/_config.yaml"},
+		},
+	}
+
+	stdout := bumper.HideSecretsWriter{Delegate: os.Stdout, Censor: sa}
+	stderr := bumper.HideSecretsWriter{Delegate: os.Stderr, Censor: sa}
+	author := fmt.Sprintf("%s <%s>", o.gitName, o.gitEmail)
+	commitsCounter := 0
+
+	for _, step := range steps {
+		committed, err := runAndCommitIfNeeded(stdout, stderr, author, step.command, step.arguments)
+		if err != nil {
+			logrus.WithError(err).Fatal("failed to run command and commit the changes")
+		}
+
+		if committed {
+			commitsCounter++
+		}
+	}
+	if commitsCounter == 0 {
+		logrus.Info("no new commits, existing ...")
+		os.Exit(0)
+	}
+
+	title := fmt.Sprintf("%s by auto-prow-job-dispatcher job at %s", matchTitle, time.Now().Format(time.RFC1123))
+	if err := bumper.GitPush(fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", o.githubLogin, string(sa.GetTokenGenerator(o.GitHubOptions.TokenPath)()), o.githubLogin, githubRepo), remoteBranch, stdout, stderr, ""); err != nil {
```

Comment:
We have `pkg/github/prcreation/` for the whole pr creation logic, any reason not to use that?

Reply:
Updated.